src/libraries/Common/src/Interop/Unix/System.Native/Interop.Aio.cs

```diff
@@ -0,0 +1,129 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct IoEvent
+        {
+            internal ulong Data;
+            internal ulong Obj;
+            internal long Res;
+            internal long Res2;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct IoControlBlock
+        {
+            internal ulong AioData;
+
+            // these fields swap for big endian
+            // https://github.com/torvalds/linux/blob/0a679e13ea30f85a1aef0669ee0c5a9fd7860b34/include/uapi/linux/aio_abi.h#L77-L83
+            private uint _swappedField_1;
```

Comment:
this is a little bit tricky: Linux Kernel has different order of these two fields depending on the endianess...

I've encapsulated that my making them private and adding internal properties with some simple logic that handles the endianess

Reply:
I wouldn't bother too much for fields that aren't used.